docs/pages/release_notes.md

```diff
@@ -49,6 +51,29 @@ Since this release, PMD will also expose any getter returning a collection of an
 
 ### 🚨 API Changes
 
+#### CLI
+
+* New exit code 5 introduced. PMD and CPD will exit now by default with exit code 5, if any recoverable error
+  (e.g. parsing exception, lexing exception or rule exception) occurred. PMD will still create a report with
+  all detected violations or duplications if recoverable errors occurred. Such errors mean, that the report
+  might be incomplete, as either violations or duplications for an entire file or for a specific rule are missing.
+  These cases can be considered as false-negatives.
+
+  In any case, the root cause should be investigated. If it's a problem in PMD itself, please create a bug report.
+
+* New CLI parameter `--no-fail-on-error` to ignore such errors and not exit with code 5. By default,
+  a build with errors will now fail and with that parameter, the previous behavior can be restored.
+  This parameter is available for both PMD and CPD.
+
+* The CLI parameter `--skip-lexical-errors` is deprecated. Use the new parameter `--[no-]--fail-on-error` instead.
+
+##### Ant
+
+* CPDTask has a new parameter `failOnError`. In controls, whether to fail the build if any recoverable errors occurred.
+  By default, the build will fail. CPD will still create a report with all detected duplications, but the report might
+  be incomplete.
+* The parameter `skipLexicalError` in CPDTask is deprecated. Use the new parameter `failOnError` instead.
+
 #### Deprecated API
```

Comment:
TODO:
* CPDConfiguration.isSkipLexicalErrors
* CPDConfiguration.setSkipLexicalErrors
* CPDTask.setSkipLexicalErrors

Reply:
Done. I think, this PR is finally ready now.